packages/buidler-etherscan/src/config.ts

```diff
@@ -2,10 +2,13 @@ import { ResolvedBuidlerConfig } from "@nomiclabs/buidler/types";
 
 import { EtherscanConfig } from "./types";
 
+// TODO: This function doesn't guarantee an EtherscanConfig object.
+// Some of our users don't use TypeScript and instead use plain JavaScript.
+// This means that the config.etherscan field could actually be anything.
```

Comment:
I actually wrote that down because I thought it might be better to check the validity of the configuration there instead of doing it in the task itself. I think this could be removed on grounds of looking to support verification without an API key though.

Reply:
I agree with Franco, there's little value on that comment.

Regarding config validation, I'd only do it for common errors, for example, a required config value not being present. Adding complex logic to validate cases that won't probably happen seems unnecessary.

That being said, we should offer plugins the possibility to augment the already existing config validation process. Once we have that, it should be simple to add very strict validations.